File: YZDEV20161107-dotnet_kevin-6ccf4b83ce002b6bcf2bb1831aed4c0b6f58f82e/20170106/OESClient/OESModel/Pagination.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OESModel
{
    public class Pagination
    {
        private int pageSize;
        private int currentPage;
        private int offest;
        private int pageCount;
        private int totalCount;
        private string otherType;

        public string SortName
        {
            set;
            get;
        }

        public string SortDirection
        {
            set;
            get;
        }

        public string OtherType
        {
            set
            {
                otherType = value;
            }
            get
            {
                if (otherType == null)
                {
                    otherType = "";
                }
                return otherType;
            }
        }

        public int PageSize
        {
            set { pageSize = value; }
            get
            {
                if (pageSize == 0)
                {
                    pageSize = 10;
                }
                return pageSize;
            }
        }

        public int CurrentPage
        {
            set { currentPage = value; }
            get
            {
                if (currentPage < 1)
                {
                    currentPage = 1;
                }
                return currentPage;
            }
        }

        public int Offset
        {
            get
            {
                return offest = (currentPage - 1) * pageSize;
            }
        }

        public int TotalCount
        {
            set;
            get;
        }

        public int PageCount
        {
            get
            {
                if (totalCount < 1)
                {
                    return pageCount = 0;
                }
                return pageCount = (totalCount - 1) / pageSize + 1;
            }
        }
    }
}

File: YZDEV20161107-kevin.wen-a0ce459eeba41e012f375d8de/YZDEV20161107-kevin.wen-a0ce459eeba41e012f375d8dee60c377131493f1/20161203/OnlineExamSystem/src/com/augmentum/oes/util/Pagination.java
/*
 * Copyright (c) 1994, 2017, Augmentum and/or its affiliates. All rights reserved.
 * AUGMENTUM PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */
package com.augmentum.oes.util;

public class Pagination {

    private static final String KEY_PAGE_SIZE = "pagination.pageSize";

    private int pageSize;
    private int pageCount;
    private int totalCount;
    private int currentPage;

    private int offset;
    private String search;
    private String sort = "DESC";

    public String getSearch() {
        return search;
    }

    public void setSearch(String search) {
        this.search = search;
    }


    public String getSort() {
        return sort;
    }

    public void setSort(String sort) {
        this.sort = sort;
    }

    public int getPageSize() {
        if(pageSize == 0){
            String size = PropertyUtil.getProperties(KEY_PAGE_SIZE);
            pageSize = Integer.parseInt(size);
        }
        return pageSize;
    }

    public void setPageSize(int pageSize) {
        this.pageSize = pageSize;
    }

    public int getPageCount() {
        if(totalCount <1) {
            pageCount = 0;
            return pageCount;
        }

        pageCount = (totalCount-1)/getPageSize()+1;
        return pageCount;
    }

    public void setPageCount(int pageCount) {
        this.pageCount = pageCount;
    }

    public int getTotalCount() {
        return totalCount;
    }

    public void setTotalCount(int totalCount) {
        this.totalCount = totalCount;
    }

    public int getCurrentPage() {
        if(currentPage <1) {
            currentPage = 1;
        }
        return currentPage;
    }

    public void setCurrentPage(int currentPage) {
        this.currentPage = currentPage;
    }

    public int getOffset() {
        offset = (getCurrentPage() -1 )* getPageSize();
        return offset;
    }

    public boolean isFirstPage() {
        if(this.getCurrentPage()<=1){
            return true;
        }
        return false;
    }

    public boolean isLastPage() {
        if(this.getCurrentPage()>=this.getPageCount()){
            return true;
        }
        return false;
    }
}
